hnn_core/tests/test_parallel_backends.py

```diff
@@ -104,16 +104,18 @@ def test_compare_across_backends():
     # test consistency across all parallel backends for multiple trials
     assert_raises(AssertionError, assert_array_equal,
                   dpls_reduced_default[0].data['agg'],
-                  dpls_reduced_default[1].data['agg'])
+                  dpls_reduced_default[1].data['agg'],
+                  err_msg='Model values across trials not equal.')
 
     for trial_idx in range(len(dpls_reduced_default)):
         # account for rounding error incured during MPI parallelization
         if dpls_reduced_mpi:
             assert_allclose(dpls_reduced_default[trial_idx].data['agg'],
                             dpls_reduced_mpi[trial_idx].data['agg'], rtol=0,
-                            atol=1e-14)
+                            atol=1e-14, err_msg='MPI values differ too much!')
         assert_array_equal(dpls_reduced_default[trial_idx].data['agg'],
-                           dpls_reduced_joblib[trial_idx].data['agg'])
+                           dpls_reduced_joblib[trial_idx].data['agg'],
+                           err_msg='Joblib values not equal to serial impl.')
```

Comment:
I got this comment wrong (will fix in next commit): the test is to make sure that two trials have _different_ model prediction values! Have updated PR description to indicate where the problem lies. No easy solution in sight. Good news is that MPI tests now pass (single-trial runs produce the expected spike times).

Reply:
Well I found a way around it, with smaller edits than I had feared